modifytags/gomodifytags.go

```diff
@@ -0,0 +1,830 @@
+package gomodifytags
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"go/ast"
+	"go/format"
+	"go/parser"
+	"go/printer"
+	"go/token"
+	"io"
+	"io/ioutil"
+	"sort"
+	"strconv"
+	"strings"
+	"unicode"
+
+	"github.com/fatih/camelcase"
+	"github.com/fatih/structtag"
+	"golang.org/x/tools/go/buildutil"
+)
+
+// structType contains a structType node and it's name. It's a convenient
+// helper type, because *ast.StructType doesn't contain the name of the struct
+type structType struct {
+	name string
+	node *ast.StructType
+}
+
+// output is used usually by editors
+type output struct {
+	Start  int      `json:"start"`
+	End    int      `json:"end"`
+	Lines  []string `json:"lines"`
+	Errors []string `json:"errors,omitempty"`
+}
+
+// Config defines how tags should be modified
+type Config struct {
+	File     string
+	Output   string
+	Quiet    bool
+	Write    bool
+	Modified io.Reader
```

Comment:
The fact that Modified is read from the `-modified` flag, and is an overlay format, seems like a rather esoteric detail of the command line util.

Perhaps we should make this field `Overlay map[string][]byte`, and keep the parsing of the archive to the command line.

(EDIT: but see comment below; perhaps we won't need to expose this API).